### PR TITLE
[Agent] support parse info from sofa hessian req

### PR DESCRIPTION
**Pull request by qq906907952**
### This PR is for:
- Agent

#### Checklist
- [x] Added unit test.
#### Backport to branches
- v6.3
- main
     

supplement fastcgi test

### CODE REVIEW

File: agent/Cargo.toml [hunk 1]
```diff
@@ -78,6 +78,7 @@ tonic = "0.8.1"
 envmnt = "0.10.4"
 wasmtime = "12.0.1"
 wasmtime-wasi = "12.0.1"
+hex = "0.4.3"
```
Comment (rvql) [inline agent/Cargo.toml:81]: 按字母排序

File: agent/src/flow_generator/protocol_logs/rpc/sofa_rpc.rs [hunk 2]
```diff
@@ -383,22 +390,50 @@ impl SofaRpcLog {
...
-            }
-
             if !sofa_hdr.new_rpc_trace_context.is_empty() {
                 info.fill_with_trace_ctx(sofa_hdr.new_rpc_trace_context);
             }
         }
+        // parse req hessian2 obj
+        if payload.len() != 0 && hdr.typ == TYPE_REQ {
+            if let Some(h) = HessianObjIterator::new(payload) {
+                while let Some((k, v)) = h.next() {
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/rpc/sofa_rpc.rs:410]: 这个应该可以直接`for (k, v) in h`吧

File: agent/src/flow_generator/protocol_logs/rpc/sofa_rpc/hessian.rs [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+        let Some(field_len) = s.read_integer(None) else {
+            return None;
+        };
+        let mut fields = vec![];
+        for _ in 0..field_len {
+            let Some(field) = s.read_string(None) else {
+                return None;
+            };
+            fields.push(field.to_string());
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/rpc/sofa_rpc/hessian.rs:95]: 为啥不`self.field_list.push`

File: agent/src/flow_generator/protocol_logs/rpc/sofa_rpc.rs (hunk 2 above)
Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/rpc/sofa_rpc.rs:410]: 改成了迭代器

File: agent/src/flow_generator/protocol_logs/rpc/sofa_rpc/hessian.rs (hunk 3 above)
Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/rpc/sofa_rpc/hessian.rs]: 改了

**Review by rvql — APPROVED** [review]

**Pull request merged**